/*
 * Decompiled with CFR 0_132.
 */
package io.reactivex.internal.operators.completable;

import io.reactivex.Completable;
import io.reactivex.CompletableObserver;
import io.reactivex.exceptions.Exceptions;
import io.reactivex.internal.disposables.EmptyDisposable;
import io.reactivex.internal.functions.ObjectHelper;
import java.util.concurrent.Callable;

public final class CompletableErrorSupplier
extends Completable {
    final Callable<? extends Throwable> errorSupplier;

    public CompletableErrorSupplier(Callable<? extends Throwable> callable) {
        this.errorSupplier = callable;
    }

    @Override
    protected void subscribeActual(CompletableObserver completableObserver) {
        Throwable throwable;
        try {
            throwable = ObjectHelper.requireNonNull(this.errorSupplier.call(), "The error returned is null");
        }
        catch (Throwable throwable2) {
            Exceptions.throwIfFatal(throwable2);
        }
        EmptyDisposable.error(throwable, completableObserver);
    }
}

